### PR TITLE
Add validation of option names and descriptions

### PR DESCRIPTION
**Pull request by Middledot**
## Summary
Add validation checks for options. 

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

File: discord/commands/core.py [hunk 1]
```diff
@@ -1610,18 +1619,18 @@ def validate_chat_input_name(name: Any, locale: Optional[str] = None):
...
     if locale not in valid_locales and locale is not None:
         raise ValidationError(
-            f"Locale {locale} is not a valid locale, in command names, "
+            f"Locale '{locale}' is not a valid locale, "
             f"see {docs}/reference#locales for list of supported locales."
         )
     if not isinstance(name, str):
         raise TypeError(
-            f"Chat input command names and options must be of type str." f"Received {name}" + f" in locale {locale}"
+            f"Command names and options must be of type str." f"Received {name}" + f" in locale {locale}"
```
Comment (krittick) [inline discord/commands/core.py:1627]:
```suggestion
            f"Command names and options must be of type str." f"Received {name} in locale {locale}"
```

File: discord/commands/core.py [hunk 2]
```diff
@@ -1630,14 +1639,14 @@ def validate_chat_input_name(name: Any, locale: Optional[str] = None):
         )
     if not 1 <= len(name) <= 32:
         raise ValidationError(
-            "Chat input command names and options must be 1-32 characters long. "
+            "Command names and options must be 1-32 characters long. "
             f"Received {name}" + f" in locale {locale}"
```
Comment (krittick) [inline discord/commands/core.py:1643]:
```suggestion
            f"Received {name} in locale {locale}"
```
Might as well change them all if we change one...

File: discord/commands/core.py [hunk 3]
```diff
@@ -1630,14 +1639,14 @@ def validate_chat_input_name(name: Any, locale: Optional[str] = None):
...
             if locale
             else ""
         )
     if not name.lower() == name:  # Can't use islower() as it fails if none of the chars can be lower. See #512.
         raise ValidationError(
-            "Chat input command names and options must be lowercase. " f"Received {name}" + f" in locale {locale}"
+            "Command names and options must be lowercase. " f"Received {name}" + f" in locale {locale}"
```
Comment (krittick) [inline discord/commands/core.py:1649]:
```suggestion
            f"Command names and options must be lowercase. Received {name} in locale {locale}"
```

File: discord/commands/core.py [hunk 4]
```diff
@@ -1646,18 +1655,18 @@ def validate_chat_input_name(name: Any, locale: Optional[str] = None):
...
     if locale not in valid_locales and locale is not None:
         raise ValidationError(
-            f"Locale {locale} is not a valid locale, in command descriptions, "
+            f"Locale '{locale}' is not a valid locale, "
             f"see {docs}/reference#locales for list of supported locales."
         )
     if not isinstance(description, str):
         raise TypeError(
-            f"Command description must be of type str. Received {description} " + f" in locale {locale}"
+            f"Command and option description must be of type str. Received {description} " + f" in locale {locale}"
```
Comment (krittick) [inline discord/commands/core.py:1663]:
```suggestion
            f"Command and option description must be of type str. Received {description} in locale {locale}"
```

File: discord/commands/core.py [hunk 5]
```diff
@@ -1646,18 +1655,18 @@ def validate_chat_input_name(name: Any, locale: Optional[str] = None):
...
             if locale
             else ""
         )
     if not 1 <= len(description) <= 100:
         raise ValidationError(
-            "Command description must be 1-100 characters long. " f"Received {description}" + f" in locale {locale}"
+            "Command and option description must be 1-100 characters long. " f"Received {description}" + f" in locale {locale}"
```
Comment (krittick) [inline discord/commands/core.py:1669]:
```suggestion
            "Command and option description must be 1-100 characters long. " f"Received {description} in locale {locale}"
```

**Review by krittick — DISMISSED** [review]

**Review by Lulalaby — DISMISSED** [review]

**Comment by Middledot** [comment]
“in locale …” is supposed to be optionally added since the main name doesn't have a locale attached, and with how the if/else is placed, description-less errors happen

image

It can be reformatted so that it adds it to the string afterwards though since that solves formatting issues

**Review by Lulalaby — APPROVED** [review]

**Comment by Lulalaby** [comment]
Good job

**Comment by Dorukyum** [comment]
The "adding locale to error message" part is done multiple times. What if we did the following:
1. Instead of raising the chosen error, set it to a var (e.g. `error = ValidationError`)
2. At the end check for the locale, add it to the error message if it's passed, and raise the variable `error` with the set message

**Comment by Dorukyum** [comment]
The "adding locale to error message" part is done multiple times. What if we did the following:
1. Instead of raising the chosen error, set it to a var (e.g. `error = ValidationError`)
2. At the end check for the locale, add it to the error message if it's passed, and raise the variable `error` with the set message

**Review by Dorukyum — APPROVED** [review]
Looks good, thanks

**Review by krittick — APPROVED** [review]

**Pull request merged**